src/state/store.ts

```diff
@@ -25,202 +21,213 @@ export interface UIState {
+    editRecipe: (recipeId: string, updates: Record<string, string | number>) => void;
     restoreRecipeDefault: (recipeId: string) => void;
     getCurrentValue: (path: string) => string | number | undefined;
     getOriginalValue: (path: string) => string | number | undefined;
-    startPacking: (
-        callback: (recipeId: string, configId: string, recipeString: string) => Promise<void>
-    ) => Promise<void>;
+    startPacking: () => Promise<void>;
+    reset: () => void;
 };
 
 export type RecipeStore = RecipeState & UIState & Actions;
 
 const initialState: RecipeState & UIState = {
-    selectedInputName: "",
     selectedRecipeId: "",
-    inputOptions: {},
     recipes: {},
     isLoading: false,
     isPacking: false,
+    packingData: EMPTY_PACKING_DATA,
 };
 
 export const useRecipeStore = create<RecipeStore>()(
     subscribeWithSelector((set, get) => ({
         ...initialState,
 
-        loadInputOptions: async () => {
+        loadAllRecipes: async () => {
             set({ isLoading: true });
             try {
-                const inputOptions = await getPackingInputsDict();
-                set({ inputOptions });
+                const recipes = await getRecipesFromFirebase();
+                set({ recipes });
+                // make an intial selection
+                const firstId = Object.keys(recipes)[0];
+                set(s => (!s.selectedRecipeId && firstId ? { selectedRecipeId: firstId } : {}));
             } finally {
                 set({ isLoading: false });
             }
         },
 
+        selectRecipe: async (recipeId) => {
+            const rec = get().recipes[recipeId];
+            // TODO: fails silently? error handling here?
+            if (!rec) return;
 
-        loadRecipe: async (recipeId) => {
-            if (get().recipes[recipeId]) return;
-            const recJson = await getFirebaseRecipe(recipeId);
-            const recStr = jsonToString(recJson);
-            set(s => ({
-                recipes: {
-                    ...s.recipes,
-                    [recipeId]: {
-                        id: recipeId,
-                        originalString: recStr,
-                        currentString: recStr,
-                        isModified: false,
-                    },
-                },
-            }));
-
+            set(s => (s.selectedRecipeId === rec.recipeId ? {} : { selectedRecipeId: rec.recipeId }));
         },
 
-        loadAllRecipes: async () => {
-            const { inputOptions, recipes, loadRecipe } = get();
+        editRecipe: (recipeId, updates) => {
+            const rec = get().recipes[recipeId];
+            if (!rec) return;
 
-            const ids = new Set<string>();
-            Object.values(inputOptions).forEach(opt => { if (opt?.recipe) ids.add(opt.recipe); });
-            const missing = [...ids].filter(id => !recipes[id]);
-            if (!missing.length) return;
+            const newEdits = { ...rec.edits };
```

Comment:
wouldn't these been the currentEdits?

Reply:
i'm surprised this isn't just one field at a time. how is a user able to send in multiple edits at once to state?